api/src/main/java/org/mandulis/mts/service/MinioStorageService.java

```diff
@@ -0,0 +1,81 @@
+package org.mandulis.mts.service;
+
+import io.minio.*;
+import org.mandulis.mts.exception.StorageException;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.InputStream;
+import java.nio.file.Path;
+
+@Service
+public class MinioStorageService implements StorageService {
+
+    private final String endpoint;
+    private final String username;
+    private final String password;
+    private final String bucketName;
+
+    MinioStorageService(
+            @Value("${minio.endpoint}") String endpoint,
+            @Value("${minio.username}") String username,
+            @Value("${minio.password}") String password,
+            @Value("${minio.bucketName}") String bucketName
```

Comment:
please add the Value annotations on the instance variables

Reply:
That would make property-based injection. Didn't we wanted constructor based? Just like in other comment?

Reply:
The timing of this injection is typically at the time of bean initialization, which usually occurs when the application context is started. Once the value is injected, it remains constant unless the application context is refreshed or the bean is re-initialized.